lib/flags.js

```diff
@@ -62,22 +58,29 @@ module.exports = {
 
     if (argv.logLevel) {
       result.logLevel = argv.logLevel;
-      result.devMiddleware = Object.assign(devMiddleware, {
+      result.devMiddleware = Object.assign(devMiddleware || {}, {
         logLevel: result.logLevel,
       });
-      result.hotClient = Object.assign(hotClient, {
+      result.hotClient = Object.assign(hotClient || {}, {
         logLevel: result.logLevel,
       });
     }
 
     if (argv.logTime) {
       result.logTime = true;
-      result.devMiddleware = Object.assign(devMiddleware, { logTime: true });
-      result.hotClient = Object.assign(hotClient, { logTime: true });
+      result.devMiddleware = Object.assign(devMiddleware || {}, {
+        logTime: true,
+      });
+      result.hotClient = Object.assign(hotClient || {}, { logTime: true });
     }
 
-    result.devMiddleware = devMiddleware;
-    result.hotClient = hotClient;
+    if (devMiddleware) {
```

Comment:
I think this lines and the below ones for hotClient cancells the assignment above inside the if `argv.loglevel` and `argv.logTime` (and maybe those cancel each other too?, can't do both logTime and logLevel)

Reply:
keen eye. had a change queued up to address that. the entire handling of `devMiddleware` and `hotClient` had to be reworked.